sound/soc/sof/intel/hda-dai.c

```diff
@@ -400,6 +400,101 @@ static const struct snd_soc_dai_ops hda_link_dai_ops = {
 	.prepare = hda_link_pcm_prepare,
 };
 
+static struct sof_ipc_dai_config* ssp_dai_config_control(struct snd_sof_dev *sdev,
+							 struct snd_sof_dai *dai)
+{
+	struct sof_ipc_dai_config *config;
+	struct snd_sof_control *scontrol;
+	struct sof_ipc_ctrl_data *cdata;
+	int number_configs = dai->number_configs;
+	int current_config = dai->current_config;
+	int index;
+
+	if (current_config > number_configs - 1) {
+		dev_err(sdev->dev, "error: dai config index out of range\n");
+		return NULL;
+	}
+
+	config = &dai->dai_config[current_config];
+
+	/* return the only possible config if only 1 config present */
+	if (number_configs < 2)
+		return config;
+
+	/* find out if there's a control associated with the dai */
+	list_for_each_entry(scontrol, &sdev->kcontrol_list, list) {
+		cdata = scontrol->control_data;
+		if (scontrol->comp_id == dai->comp_dai.comp.id) {
+			/* find out if this control is meant for choosing dai config */
+			if (cdata->cmd == SOF_CTRL_CMD_ENUM) {
+				index = cdata->chanv[0].value;
+				/* check that the control value is in range and for ssp only */
+				if (index < number_configs &&
+				    dai->dai_config[index].type == SOF_DAI_INTEL_SSP) {
```

Comment:
what if tomorrow Seppo wants to try multiple configs for microphones? What's so special about the SSP?

Reply:
nothing, I just didn't want to touch other than SSP and keep the original functionality for the other DAI's. IMHO all dai's should support the multiple configs. This is just trying to minimize the damage if someone tries to do some strange configuration for DAI's that don't currently save multiple configs.

Reply:
why do we need to check, you dont set the BE hw_params for DMICs anyway. So its not like you can even get here for DMICs

Reply:
ok, removing this